core/src/main/scala/chisel3/Data.scala

```diff
@@ -372,6 +374,12 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
     }
   }
 
+  // probeInfo only exists if this is a probe type
+  private var _probeInfoVar:      ProbeInfo = null
+  private var _probeInfo:         Option[ProbeInfo] = Option(_probeInfoVar)
+  private[chisel3] def probeInfo: Option[ProbeInfo] = _probeInfo
+  private[chisel3] def probeInfo_=(probeInfo: Option[ProbeInfo]) = _probeInfo = probeInfo
```

Comment:
This logic isn't quite right, we have `_probeInfoVar` but aren't actually using it (defeating the purpose).
```suggestion
  private var _probeInfoVar:      ProbeInfo = null
  private def _probeInfo:         Option[ProbeInfo] = Option(_probeInfoVar)
  private[chisel3] def probeInfo: Option[ProbeInfo] = _probeInfo
  private[chisel3] def probeInfo_=(probeInfo: Option[ProbeInfo]) = _probeInfoVar = probeInfo.getOrElse(null)
```